### PR TITLE
Add cover letter functionality

### LINKED ISSUE #614: [FEATURE]: Usage mode using a specific resume AND cover letter
### Feature summary

Allow users to specify a specific cover letter for applications

### Feature description

Currently, the program can be run using a specific resume; this feature would extend that functionality to include a specific cover letter as well

### Motivation

The same reasons I want to use a specific resume, I don't like the template or AI-developed content or want greater consistency / fewer errors in AI-generation

### Alternatives considered

NA

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by yusefes**
Fixes #614

Add functionality to specify a specific cover letter for job applications.

* **main.py**
  - Add `cover_letter` option to the `main` function to specify a cover letter file.
  - Update `FileManager.file_paths_to_dict` method to handle the `cover_letter` file.
  - Pass the `cover_letter` file to the `create_and_run_bot` function.

* **src/aihawk_easy_applier.py**
  - Update `_handle_upload_fields` method to handle the `cover_letter` file.
  - Add logic to upload cover letter from the specified path if it exists.

* **src/job_application_profile.py**
  - Add `cover_letter` attribute to the `JobApplicationProfile` class.
  - Update `__init__` method to initialize the `cover_letter` attribute.

* **data_folder_example/plain_text_resume.yaml**
  - Add `cover_letter` field with example content.

* **data_folder/plain_text_resume.yaml**
  - Add `cover_letter` field.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/feder-cr/Auto_Jobs_Applier_AIHawk/issues/614?shareId=b4ec2c66-5d15-4e31-891c-7601b82ce8e7).

### CODE REVIEW

File: src/aihawk_easy_applier.py [hunk 1]
```diff
@@ -434,7 +434,11 @@ def _handle_upload_fields(self, element: WebElement, job) -> None:
                     self._create_and_upload_resume(element, job)
             elif 'cover' in output:
                 logger.debug("Uploading cover letter")
-                self._create_and_upload_cover_letter(element, job)
+                if job.cover_letter_path is not None and os.path.exists(job.cover_letter_path):
```
Comment (surapuramakhil) [inline src/aihawk_easy_applier.py:437]: move this check to method something like  if_coverletter_exits or something silimar. method name should be check objective.

File: main.py [hunk 2]
```diff
@@ -136,7 +136,7 @@ def validate_data_folder(app_data_folder: Path) -> tuple:
         return (app_data_folder / 'secrets.yaml', app_data_folder / 'config.yaml', app_data_folder / 'plain_text_resume.yaml', output_folder)
 
     @staticmethod
-    def file_paths_to_dict(resume_file: Path | None, plain_text_resume_file: Path) -> dict:
+    def file_paths_to_dict(resume_file: Path | None, plain_text_resume_file: Path, cover_letter_file: Path | None = None) -> dict:
```
Comment (surapuramakhil) [inline main.py:139]: Boy Scout Code Principle

Can you move FileManager to a different file maybe file_manager.py

rename method file_paths_to_dict to something meaning full. refactor method if possible

File: src/job_application_profile.py [hunk 3]
```diff
@@ -169,6 +170,15 @@ def __init__(self, yaml_str: str):
             logger.error(f"An unexpected error occurred while processing salary_expectations: {e}")
             raise RuntimeError("An unexpected error occurred while processing salary_expectations.") from e
 
+        # Process cover_letter
```
Comment (surapuramakhil) [inline src/job_application_profile.py:173]: Appriciate if you can refactor the code - DRY principle

you see code patterns are repeated. can you remove those repeated code patterns using method calls

File: src/aihawk_easy_applier.py (hunk 1 above)
Comment (surapuramakhil) [reply src/aihawk_easy_applier.py:437]: I guess you are missing case where cover is in plain text resume.yaml

**Review by surapuramakhil — CHANGES_REQUESTED** [review]
this one is important - https://github.com/feder-cr/Auto_Jobs_Applier_AIHawk/pull/628/files#r1818228664

Other requests are for improving this code base. Appricate if those were addressed.

@yusefes good work

**Comment by feder-cr** [comment]
@yusefes resolve the conflicts

**Pull request closed without merging**